UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetForLargeFile.cs

```diff
@@ -61,23 +60,37 @@ public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAMediaTxtF
 
         [Test]
         [Category("QCOnlyTest")]
-        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnINFOFolderIsGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnINFOFolderWithFilesIsGenerated()
         {
+            string[] infoFolderFiles = { "AVCS-User-Guide.pdf", "ENC TandP NM status.pdf", "End-User-Licence-Agreement-for-ADMIRALTY-digital-data-services.pdf", "Important Information for AVCS users.pdf" };
             foreach (string folderPath in DownloadedFolderPath)
             {
+                //To verify the INFO folder exists
                 bool checkFolder = FssBatchHelper.CheckforFolderExist(folderPath, Config.POSConfig.LargeExchangeSetInfoFolderName);
                 Assert.IsTrue(checkFolder, $"Folder not Exist in the specified folder path :");
+
+                //To verify the files under INFO folder exists
+                foreach (string infoFile in infoFolderFiles)
+                {
+                    bool checkFile = FssBatchHelper.CheckforFileExist(Path.Combine(folderPath, Config.POSConfig.LargeExchangeSetInfoFolderName),infoFile);
+                    Assert.IsTrue(checkFile, $"{infoFile} does not Exist in the specified folder path.");
+                }
             }
         }
 
         [Test]
         [Category("QCOnlyTest")]
-        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnADCFolderIsGenerated()
+        public void WhenICallExchangeSetApiWithMultipleProductIdentifiers_ThenAnADCFolderWithFilesIsGenerated()
         {
             foreach (string folderPath in DownloadedFolderPath)
             {
+                //To verify the ADC folder exists
                 bool checkFolder = FssBatchHelper.CheckforFolderExist(Path.Combine(folderPath, Config.POSConfig.LargeExchangeSetInfoFolderName), Config.POSConfig.LargeExchangeSetAdcFolderName);
-                Assert.IsTrue(checkFolder, $"Folder not Exist in the specified folder path :");
+                Assert.IsTrue(checkFolder, $"Folder not Exist in the specified folder path.");
+
+                //To verify that the files exists under ADC folder
+                int fileCount = Directory.GetFiles(Path.Combine(folderPath, Config.POSConfig.LargeExchangeSetInfoFolderName, Config.POSConfig.LargeExchangeSetAdcFolderName),"*.*",SearchOption.TopDirectoryOnly).Length;
+                Assert.IsTrue(fileCount > 0, $"File count is {fileCount} in the specified folder path.");
```

Comment:
this code is duplicated for this test and the previous one. you need to create a test method that uses the same code but can be reutilized for the two tests

Reply:
As per our discussion no need to fix this.

Thanks,
Mayuresh Gawde